app/models/evaluator_submission_assignment.rb

```diff
@@ -73,6 +73,6 @@ def assigned_evaluation_status
   end
 
   def update_submission_evaluation_status
-    submission.update_submission_evaluation_status
+    submission.save
```

Comment:
call the Service directly instead of relying on the before_save callback:
```suggestion
    EvaluationStatusService.update_evaluation_status(submission)
```